### PR TITLE
Add a CI presubmit that renders the documentation.

### PR DESCRIPTION
**Pull request by hawkinsp**

### CODE REVIEW

**Review by yashk2810 — COMMENTED** [review]
Why do we want to block CLs on this?

or does this just make the errors a bit better than the very confusing sphinx site without blocking CLs?

File: .github/workflows/ci-build.yaml [hunk 1]
```diff
@@ -124,6 +124,7 @@ jobs:
 
 
   documentation:
+    name: Documentation - test code snippets
```
Comment (yashk2810) [inline .github/workflows/ci-build.yaml:127]: For this, will copybara need to be changed?

Comment (hawkinsp) [reply .github/workflows/ci-build.yaml:127]: No. Our copybara config does not actually check the documentation build (either of them).

**Comment by hawkinsp** [comment]
> Why do we want to block CLs on this?
> 
> or does this just make the errors a bit better than the very confusing sphinx site without blocking CLs

The explicit goal is to not let you submit PRs that break the documentation build.

**Comment by yashk2810** [comment]
> The explicit goal is to not let you submit PRs that break the documentation build

I guess that's reasonable. But why didn't we block on this before then? Would be nice to dig up that reason. Also it is very hard to fix docs builds from CLs. The errors are also not legible. 

But if you feel strongly then LGTM.

**Comment by hawkinsp** [comment]
> > The explicit goal is to not let you submit PRs that break the documentation build
> 
> I guess that's reasonable. But why didn't we block on this before then? Would be nice to dig up that reason. Also it is very hard to fix docs builds from CLs. The errors are also not legible.
> 
> But if you feel strongly then LGTM.

We used to block on this. But we were running the presubmit in readthedocs, and we ran out of readthedocs capacity. So we turned it off rather than block people. But I don't think that applies if the blocking part runs jn github actions.

Note: at the moment I'm trying using a free runner. If that's too slow, we can use a larger runner.

**Comment by yashk2810** [comment]
I see okay.

**Review by yashk2810 — APPROVED** [review]

File: .github/workflows/ci-build.yaml [hunk 2]
```diff
@@ -159,3 +160,39 @@ jobs:
       run: |
         pytest -n auto --tb=short docs
         pytest -n auto --tb=short --doctest-modules jax --ignore=jax/experimental/jax2tf --ignore=jax/_src/lib/mlir --ignore=jax/interpreters/mlir.py --ignore=jax/_src/iree.py --ignore=jax/experimental/array_serialization --ignore=jax/collect_profile.py
+
+
+  documentation_render:
+    name: Documentation - render documentation
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
```
Comment (jakevdp) [inline .github/workflows/ci-build.yaml:168]: Is this long enough?

**Review by jakevdp — APPROVED** [review]

File: .github/workflows/ci-build.yaml (hunk 2 above)
Comment (hawkinsp) [reply .github/workflows/ci-build.yaml:168]: Seems to be. It took 5 minutes on the latest run from PR.

If it's longer than that, my next step would be to make the runner a larger one. Let's try it out!

**Pull request merged**